### ISSUE TITLE
how to use customize resource file with faceeffect?

### ISSUE THREAD

**Issue by bugmany** [google/mediapipe#1789]
In project FaceEffect, with effect files(such as facepaint and glasses) pngblob and binarypb, ran ok, but these resources are intergraded into project when built, so i want whether divide these resources, and load its when run the project.

i've research the sources file "face_geometry/effect_renderer_calculator.cc", with functions: 
ReadContentBlobFromFile(), 
ReadMesh3dFromFile(),
i've parsed resource's file path and instead of with customize file path, but failed to render.

**Comment by sgowroji** [comment]
Hi @bugmany May be [this](https://google.github.io/mediapipe/solutions/face_mesh#effect-renderer) and https://github.com/google/mediapipe/blob/master/mediapipe/modules/face_geometry/effect_renderer_calculator.cc can address your queries well. Let me know if you have any further queries. Thanks!

**Comment by bugmany** [comment]
@sgowroji Thanks for you reply, i've read the document [face-effect-renderer](https://google.github.io/mediapipe/solutions/face_mesh#effect-renderer), here introduced how to render face effect, didnot refer to customize resource files.

**Comment by sgowroji** [comment]
I have assigned to @kostyaby he may guide you about it. But take a look at this [issue](https://github.com/google/mediapipe/issues/1155) for better understanding.

**Comment by bugmany** [comment]
Hi @kostyaby, Could you give me advice?

**Comment by kostyaby** [comment]
Hey @bugmany,

Let me rephrase your question to make sure I understand it correctly: you wanted to enable `FaceGeometryEffectRendererCalculator` assets to be loaded not from an app binary, but from some other path on Android / iOS and you had troubles doing that, correct? 

If that's the case, could you please share the following details so I have enough context: 

- Did you place those external assets at some file system location? 
- Were you able to confirm those assets are indeed located at that path through some OS tool?
  - On Android you can check that via `adb shell` and then navigate the file system of your device to make sure the file is located at the correct path;
  - On iOS you should be able to check via `fileManager fileExistsAtPath` ([docs](https://developer.apple.com/documentation/foundation/nsfilemanager/1410277-fileexistsatpath?language=objc))
- Did the path you replaced, for example, [here](https://github.com/google/mediapipe/blob/master/mediapipe/graphs/face_effect/face_effect_gpu.pbtxt#L114) is absolute or relative?
  - `FaceGeometryEffectRendererCalculator` relies on [mediapipe::PathToResourceAsFile](https://github.com/google/mediapipe/blob/350fbb2100ad531bc110b93aaea23d96af5a5064/mediapipe/util/resource_util.h#L25-L42) to help resolving relative paths;
  - On both [Android](https://github.com/google/mediapipe/blob/350fbb2100ad531bc110b93aaea23d96af5a5064/mediapipe/util/resource_util_android.cc#L35-L38) and [iOS](https://github.com/google/mediapipe/blob/350fbb2100ad531bc110b93aaea23d96af5a5064/mediapipe/util/resource_util_apple.cc#L44-L47), absolute paths are kept as-is;
  - However, relative path resolution is platform-dependent so if that's you are specifying, then I'd expect it to be the source of your troubles and would recommend you to switch to absolute paths if that's possible

In general, `FaceGeometryEffectRendererCalculator` is more of a sample renderer and it is intended mainly for example purposes. It's great if people can get more out of it and I'm happy to add minor features to it (like, maybe passing asset paths as side-packets so you don't have to hard-code it into the graph) - however, please be mindful of its "minimum viable product" nature :) If you need even more features from the calculator, please don't hesitate to just fork it and add those features as your project requires

Hopefully, I got your question right and this reply was helpful. Please, let me know if that's not the case and I'll provide more guidance

**Comment by bugmany** [comment]
Hi @kostyaby, i'll try it, thanks.
